### DISCUSSION TITLE
User Input

### DISCUSSION THREAD

**Discussion by johnlindquist** [johnlindquist/kit#186] [Guide]
### `await arg()`

```js
let input = await arg()
```

Launches the prompt, waits for user input.

### `await env("SOME_TOKEN")`

```js
let SOME_TOKEN = await env("SOME_TOKEN")
```

Launches the prompt if var isn't found in `.env`. Waits for user input. Automatically stores var in `.env`.

### `await drop()`

```js
let dropData = await drop()
```

Launches the prompt as a drop target.

### `await hotkey()`

```js
let keyInfo = await hotkey()
```

Launches the prompt and waits for a key combination.

**Comment by gr2m** [comment]
> await env("SOME_TOKEN")

Is there a way to pass in an asynchronous function in case the `env` is not set? Example:

```js
const token = await env("GITHUB_TOKEN", { get: doGitHubOAuthDeviceFlow({ clientId, onVerification }) })
```

Or as an alternative, is there a way to explicitly set an `env` variable?

**Reply by johnlindquist** [reply]
You can manually check:
```js
if(env.GITHUB_TOKEN)
```

And manually set:
```js
await cli("set-env-var", "GITHUB_TOKEN", value);
```

**Reply by gr2m** [reply]
not sure if this is a bug or a feature, but

```js
await cli("set-env-var", "GITHUB_TOKEN_NO_SCOPES", token);
env.GITHUB_TOKEN_NO_SCOPES // undefined
```

I expected the `env` object to be live updated when I use the `set-env-var` command

**Reply by johnlindquist** [reply]
Ah, good catch. It's updating the `.env` file, but not re-reading the file/updating the env object. I'll get that fix in tonight too. Thanks!

**Comment by TreTuna** [comment]
Is there an easy way to take a multi-line input from a user at the moment?

**Reply by johnlindquist** [reply]
@TreTuna I just released beta.18 tonight (I'll write up some docs tomorrow).

You can now do:
```js
let text = await textarea()

await arg("Here's what you wrote", text)
```

**Reply by TreTuna** [reply]
Just tried it out and it's exactly what I was looking for, thanks!

**Reply by kentcdodds** [reply]
Ummm... So how do I as a user of the textarea actually submit what I've typed?

**Reply by johnlindquist** [reply]
@kentcdodds 

cmd + s submits
cmd + w cancels

(that's the default placeholder text)

Same with the `editor` component:

```js
await editor()
```
